### PR TITLE
Improvements in compilation database parsing

### PR DESCRIPTION
**Pull request by oxisto**
This adds a new option to select a specific component and also improves heuristics to detect components in compilation databases generated by bear

### CODE REVIEW

**Review by konradweiss — APPROVED** [review]

File: cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+LibrariesTest.kt
```
Comment (KuechA) [inline cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/.gitignore:1]: Intentional or leftover?

File: cpg-language-cxx/build.gradle.kts [hunk 2]
```diff
@@ -52,4 +52,5 @@ dependencies {
     api(libs.eclipse.cdt.core)
 
     testImplementation(libs.junit.params)
+    testImplementation(project(":cpg-analysis"))
```
Comment (KuechA) [inline cpg-language-cxx/build.gradle.kts:55]: The checked-in test shouldn't need this dependency

**Review by KuechA — APPROVED** [review]

File: cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/.gitignore (hunk 1 above)
Comment (oxisto) [reply cpg-language-cxx/src/test/kotlin/de/fraunhofer/aisec/cpg/frontends/cxx/.gitignore:1]: This is somewhat intentional

File: cpg-language-cxx/build.gradle.kts (hunk 2 above)
Comment (oxisto) [reply cpg-language-cxx/build.gradle.kts:55]: They do. we are using `evaluate()`

**Pull request merged**